provision/provision-ssh

```diff
@@ -10,6 +10,20 @@ scriptPath="provision/provision.sh"
 targetPath="/home/ubuntu/provision.sh"
 bastionIP=$(terraform output bastion_ip)
 
-scp -o StrictHostKeyChecking=no -i ${keyPath} $scriptPath ubuntu@$bastionIP:$targetPath
 
+#Wait until SSH on Bastion server is working
+echo "Attempting to SSH to Bastion server..."
+index=1
+set +x
+while (( $index != 0 ))
+do
+  scp -o StrictHostKeyChecking=no -i ${keyPath} $scriptPath ubuntu@$bastionIP:$targetPath
+  case $? in
+    (0) echo "${index}> Success"; break ;;
+    (*) echo "${index}> Bastion SSH server not ready yet..." ;;
+  esac
+  sleep 5
+  ((index+=1))
```

Comment:
I like that approach, just something to keep it from failing indefinitely, so we can tell the difference between the server failed to launch and we can’t ssh, and it’s just starting up still.

> On May 4, 2015, at 5:54 PM, Chris Weibel <notifications@github.com> wrote:
> 
> In provision/provision-ssh <https://github.com/cloudfoundry-community/terraform-openstack-cf-install/pull/17#discussion_r29627851>:
> 
> >  
> > +#Wait until SSH on Bastion server is working
> > +echo "Attempting to SSH to Bastion server..."
> > +index=1
> > +set +x
> > +while (( $index != 0 ))
> > +do
> > +  scp -o StrictHostKeyChecking=no -i ${keyPath} $scriptPath ubuntu@$bastionIP:$targetPath
> > +  case $? in
> > +    (0) echo "${index}> Success"; break ;;
> > +    (*) echo "${index}> Bastion SSH server not ready yet..." ;;
> > +  esac
> > +  sleep 5
> > +  ((index+=1))
> This is code I originally got from Wayne - the point was to simply increment how many times the scp was attempted before successful. We could simply change it to
> 
> while (( $index <= 10 ))
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/cloudfoundry-community/terraform-openstack-cf-install/pull/17/files#r29627851>.
>